File: src/main/java/io/vertigo/shiny/models/input/multiselection/ShinyMultiSelection.java
package io.vertigo.shiny.models.input.multiselection;

import java.util.List;
import java.util.Set;
import java.util.UUID;
import java.util.stream.Collectors;

import io.vertigo.shiny.models.ShinyElement;

public record ShinyMultiSelection(
		UUID id,
		String title,
		List<String> options,
		Set<Integer> selectedIndices) implements ShinyElement {

	public List<String> getSelectedOptions() {
		return selectedIndices.stream()
				.sorted()
				.map(options::get)
				.collect(Collectors.toList());
	}
}

File: banshee/src/models/form/multiselection/multiselection/ShinyMultiSelection.ts
import { ShinyElement } from '../../../ShinyElement';
import { ShinyState } from '../../../ShinyState';

export class ShinyMultiSelection implements ShinyElement {
  id?: string;
  shinyType: string = 'multiSelection';
  state?: ShinyState;

  title: string;
  options: string[];
  selectedIndices: Set<number>;

  constructor(
    title: string,
    options: string[],
    selectedIndices: Set<number>,
    id?: string,
    state?: ShinyState
  ) {
    this.title = title;
    this.options = options;
    this.selectedIndices = selectedIndices;
    this.id = id;
    this.state = state;
  }

  getSelectedOptions(): string[] {
    return Array.from(this.selectedIndices)
      .sort((a, b) => a - b)
      .map(index => this.options[index]);
  }
}
